homeassistant/components/alexa/smart_home.py

```diff
@@ -383,12 +377,84 @@ def get_property(self, name):
             raise _UnsupportedProperty(name)
 
         unit = self.entity.attributes[CONF_UNIT_OF_MEASUREMENT]
+        temp = self.entity.state
+        if self.entity.domain == climate.DOMAIN:
+            temp = self.entity.attributes.get(
+                climate.ATTR_CURRENT_TEMPERATURE, 0)
         return {
-            'value': float(self.entity.state),
+            'value': float(temp),
             'scale': API_TEMP_UNITS[unit],
         }
 
 
+class _AlexaThermostatController(_AlexaInterface):
+    def name(self):
+        return 'Alexa.ThermostatController'
+
+    def properties_supported(self):
+        properties = []
+        supported = self.entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE:
+            properties.append({'name': 'targetSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_LOW:
+            properties.append({'name': 'lowerSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_HIGH:
+            properties.append({'name': 'upperSetpoint'})
+        if supported & climate.SUPPORT_OPERATION_MODE:
+            properties.append({'name': 'thermostatMode'})
+        return properties
+
+    def properties_retrievable(self):
+        return True
+
+    def get_property(self, name):
+        if name == 'thermostatMode':
+            mode = self.entity.state.upper().replace('IDLE', 'OFF')
```

Comment:
Please create a proper dictionary mapping Home Assistant modes to Alexa modes.

Reply:
We should make sure to only support the modes that are defined in `climate/__init__.py`.

Reply:
There are a couple of standard states (e.g. climate.STATE_FAN_ONLY) that don't affect the thermostat setpoints. I think mapping these to `OFF` is appropriate but we can still return an error for unrecognised modes.